spec/models/repository_directory_spec.rb

```diff
@@ -26,30 +26,42 @@
       let(:dirname) { "#{repository_directory.name}-other" }
       let(:filename) { 'file' }
       let(:filepath) { File.join(dirname, filename) }
-      before do
-        repository.save_file_only(Tempfile.new('tempfile').path, filepath, 'message', user)
-      end
 
-      it 'should fail when path is the directory' do
-        repository_directory.name = dirname
-        expect(repository_directory.valid?).to be_falsy
+      let(:save_operation) do
+        file = Tempfile.new('tempfile').path
+        repository.save_file_only(file, filepath, 'message', user)
       end
```

Comment:
This is a funny diff highlighting. Only `do`, `end` and `=` are the same. Nice, github! :D